src/EFCache/CachingReader.cs

```diff
@@ -56,7 +56,18 @@ public override int FieldCount
 
         public override bool GetBoolean(int ordinal)
         {
-            return (bool)GetValue(ordinal);
+            object value = GetValue(ordinal);
+            switch (value)
+            {
+                case bool boolValue:
+                    return boolValue;
+                case int intValue:
+                    return intValue != 0;
+                case long longValue:
+                    return longValue != 0;
+                default:
+                    throw new NotImplementedException($"{value.GetType()}");
```

Comment:
The reason why I suggested blindly casting to `long` in the comment is that it will work with `byte`, `short` and other integral types. I don't think it makes sense to enumerate all these types here and I am fine with `InvalidCastException` for non-bool, non-integral values.

Reply:
I have committed this change, but then the test `GetBoolean_converts_numeric_values_for_column` fails for the `int` column.

Reply:
I see. Probably because it's boxed. Don't worry about this then.